config/src/config/network_config.rs

```diff
@@ -340,58 +340,82 @@ impl Default for RateLimitConfig {
     }
 }
 
-pub type TrustedPeerSet = HashMap<PeerId, TrustedPeer>;
+pub type PeerSet = HashMap<PeerId, Peer>;
+
+// TODO: Combine with RoleType?
+/// Represents the Role that a peer plays in the network ecosystem rather than the type of node.
+/// Determines how nodes are connected to other nodes, and how discovery views them.
+///
+/// Rules for upstream nodes via Peer Role:
+///
+/// Validator -> Always upstream if not Validator else P2P
+/// PreferredUpstream -> Always upstream, overriding any other discovery
+/// ValidatorFullNode -> Always upstream for incoming connections (including other ValidatorFullNodes)
+/// Upstream -> Upstream, if no ValidatorFullNode or PreferredUpstream.  Useful for initial seed discovery
+/// Unknown -> Undiscovered peer, likely due to a non-mutually authenticated connection always downstream
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub enum PeerRole {
```

Comment:
Yes, and I believe I got into a conversation about that to add them because of that usecase, didn't get it I guess :P

The next pieces work with the integration with the applications, I'll add them in.